chunkenc/xor.go

```diff
@@ -384,3 +384,42 @@ func (it *xorIterator) readValue() bool {
 	it.numRead++
 	return true
 }
+
+// MergeChunks vertically merges a and b, i.e., if there is any sample
+// with same timestamp in both a and b, the sample in a is discarded.
+func MergeChunks(a, b Chunk) (*XORChunk, error) {
```

Comment:
Can we reuse the `Chunk` interface and allow streaming? Basically something like `MergeChunks(a, b Chunk) Iterator` 

Example of reusing "iterator" pattern you can see here: https://github.com/improbable-eng/thanos/blob/master/pkg/store/storepb/custom.go#

Reply:
After we merge 2 chunks, we would want a chunk and not iterator, as we want to replace the old slice of chunks with non overlapping chunks.

`Chunk` interface has methods `Bytes()`, `NumSamples()`, which I believe can be known only after completely merging the chunks.

I will check if I can make it streaming in some other ways (during compaction, merge only when accessing chunks and not while getting chunks).

Reply:
Yup, as we need merging only during compaction, and as `WriteChunks(...)` uses only `Bytes()` function to write out the chunk, I think we don't need streaming.

Reply:
@bwplotka thoughts on this?
PS: Streaming is present for vertical query merge. But feels like it's not required for compaction.